src/vec/from_stream.rs

```diff
@@ -0,0 +1,25 @@
+use crate::stream::{FromStream, IntoStream, Stream};
+
+use std::pin::Pin;
+
+impl<T: Send> FromStream<T> for Vec<T> {
```

Comment:
Instead of adding this impl for `Vec<T>`, I wonder if it'd make more sense to make a blanket impl of this type:

```rust
impl<T, I> FromStream<T> for I
where
    I: FromIterator<Item = T>,
{
    // ...
}
```

We could make a similar blanket impl with `IntoStream`, too.

Reply:
Hmmm, yeah that's not a bad idea! Would substantially save us on the impls (e.g. half the list of #129) -- I'll give this a shot!

Reply:
Talked in person about this, and the current approach is similar to Rayon which seems like a good indicator that it's worth doing.